redisenterprise/datadog_checks/redisenterprise/check.py

```diff
@@ -0,0 +1,322 @@
+import sys
+from datetime import datetime, timedelta
+
+import requests
+
+from datadog_checks.base import AgentCheck, ConfigurationError
+
+# from typing import Any
+
+EVENT_TYPE = SOURCE_TYPE_NAME = 'redisenterprise'
+
+
+class RedisenterpriseCheck(AgentCheck):
+    """RedisenterpriseCheck attempts to connect to the cluster and ensure the node is the master node"""
+
+    def __init__(self, name, init_config, instances):
+        super(RedisenterpriseCheck, self).__init__(name, init_config, instances)
+        # Set this to two minutes ago which may cause duplicates but we need to get everything in the case of failover
+        self.last_event_timestamp_seen = datetime.utcnow() - timedelta(0, 120)
+
+    def _timestamp(self, date):
+        """ Allows us to return an epoch time stamp if we use python2 or python3 """
+        if sys.version_info[0] < 3 or sys.version_info[1] < 4:
+            import time
+
+            return int(time.mktime(date.timetuple()))
+        else:
+            return int(date.timestamp())
+
+    def check(self, instance):
+        host = instance.get('host')
+        port = instance.get('port', 9443)
+        event_limit = instance.get('event_limit', 100)
+        user = instance.get('user')
+        password = instance.get('password')
+        timeout = instance.get('timeout', 10)
+        verifyssl = instance.get('verify_ssl', False)
+        is_mock = instance.get('is_mock', False)
+        service_check_tags = instance.get('tags', [])
+        auth = (user, password)
+
+        if not host or not user or not password:
+            raise ConfigurationError('Configuration error, please fix redisenterprise.yaml')
+
+        try:
+
+            # noop if we are not the cluser master
+            if self._check_follower(host, port, timeout, auth, verifyssl, is_mock):
+                self.last_timestamp_seen = datetime.utcnow()
+                pass
+
+            # add the cluster FQDN to the tags
+            fqdn = self._get_fqdn(host, port, timeout, auth, verifyssl, service_check_tags)
+            service_check_tags.append('cluster:{}'.format(fqdn))
+
+            # collect the license data
+            fqdn = self._get_license(host, port, timeout, auth, verifyssl, service_check_tags)
+
+            # collect the node data
+            self._get_nodes(host, port, timeout, auth, verifyssl, service_check_tags)
+
+            # grab the DBD ID to name mapping
+            bdb_dict = self._get_bdb_dict(host, port, timeout, auth, verifyssl, service_check_tags)
+            self._get_bdb_stats(host, port, timeout, auth, verifyssl, bdb_dict, service_check_tags)
+            self._shard_usage(bdb_dict, service_check_tags, host)
+
+            # collect the events from the API - we set the timeout higher here
+            self._get_events(host, port, timeout, auth, verifyssl, bdb_dict, service_check_tags, event_limit)
+
+            self.service_check(
```

Comment:
Service checks should be documented in the service checks json

Reply:
@hithwen documented